framework/src/dm/dm_lwm2m.c

```diff
@@ -164,7 +164,7 @@ int dm_lwm2m_get_server_address(char *server_ipAddr)
 		fprintf(stderr, "%s: No data\n", __FUNCTION__);
 		return DM_ERROR_NO_DATA;
 	}
-	strncpy(server_ipAddr, g_dm_context.server_info.ipAddress, strlen(g_dm_context.server_info.ipAddress));
+	strncpy(server_ipAddr, g_dm_context.server_info.ipAddress, strlen(g_dm_context.server_info.ipAddress) + 1);
```

Comment:
isn't is safer to assign '\0' explicitly to server_ipAddr, as you mention, server_info.ipAddress might not have assigned it at its end? Please ignore if not the case.

Reply:
as 'g_dm_context' is global it has '\o' and so is 'dm_context'. So, strlen + 1 shall be enough!

Reply:
alright, please ignore my commens in that case.

Reply:
server_ipAddr is a parameter from application, it can't guarantees server_ipAddr's length is greater than ipAddress..